[QoL] Slight Resize and Reallocate to IV graph in new game screen

tomasrggm opened the pull request:
<!-- Make sure the title includes categorization (i.e. [Bug], [QoL], [Localization]) -->
<!-- Make sure that this PR is not overlapping with someone else's work -->
<!-- Please try to keep the PR self-contained (and small) -->

## What are the changes?
This commit slightly resizes the iv graph and repositions it so as to not overlap the moves segment. The text was also resized and repositioned so as to not interfere with the graph or the other components in the stats container using the custom styling options available.

## Why am I doing these changes?
Currently the IV chart and its labels needlessly clutter the stats container in a new playthrough screen for when you wish to see stats and moves at the same time. This change should make it less inconvenient if you wish to see both at the same time.

## What did change?
-add offsetValues for x and y coordinates for labels
-Change labels position, and labels size according to the showDiff boolean (using it as an indicator for post capture or ivtoggle)
-Offset labels so they don't interfere with chart or moves
-Reduce moves container to be in line with egg moves

### Screenshots/Videos
Before the change:
![image](https://github.com/pagefaultgames/pokerogue/assets/44369723/2bea32cb-583c-4375-bf71-2d0be020b40f)
After the change:
![image](https://github.com/pagefaultgames/pokerogue/assets/44369723/46652055-74a8-4455-9aea-a186a4d9b598)


## How to test the changes?
Start a new game and toggle IV in any pokemon

## Checklist
- [X] There is no overlap with another PR?
- [X] The PR is self-contained and cannot be split into smaller PRs?
- [X] Have I provided a clear explanation of the changes?
- [X] Have I tested the changes (manually)?
    - [X] Are all unit tests still passing? (`npm run test`)
- [X] Are the changes visual?
  - [X] Have I provided screenshots/videos of the changes?

sirzento commented:
Good idea but I feel like the text is a bit hard to read now. Imo just hiding the moveset would be enough.

tomasrggm commented:
I see your point. Although I prefer having both information presented to me at the same time. If there was a way to better outline the text or increase the shadow so it makes a contrast for the words to be more perceptible would be nice. I'm also fiddling with larger words to see how much i can squeeze out of the container (here's with 58 px as an example):
![image](https://github.com/pagefaultgames/pokerogue/assets/44369723/a4dfcc9b-b7f2-46ec-a232-2d2286a94204)

tomasrggm commented:
Another idea is maybe offsetting the already existing shadows in the font by the x coordinate? As I move the words around that see that the shadow changed. That could help

sirzento commented:
Yeah resizing the moveset to the same size as the egg moves would be a good idea if you still want to display them. This will enable the graph to be a bigger size but then the only change would be resizing the moveset because I think the original size would fit.

Also test other languages since the text could be longer.

tomasrggm commented:
Changed the PR according to feedback. The moveset container is smaller and reverted the changes done to the chart for the most part. Also tested with other languages and everything seems to be in order with the exception of some asian languages because for some reason their characters are absolutely massive for some reason (Don't think my changes had any impact in this regard).
Here is what is looks like right now:
![image](https://github.com/pagefaultgames/pokerogue/assets/44369723/997ecf3d-f85e-40a2-b6e7-2e71f93b429d)

tomasrggm commented:
Also, just now realized that doing fontStyle: 62 is not the same as fontStyle: "62px". The way I was doing it was messing up the shadows. Seems to be in order now:
![image](https://github.com/pagefaultgames/pokerogue/assets/44369723/5f4af96a-4c97-40db-943e-9fcced2ef631)

bennybroseph commented:
It's better to create a new font style entirely rather than manipulating stuff to make one fit.

tomasrggm commented:
Isn't that excessive in this instance? Why even have the ability for custom properties if we're being so strict with it? All that was done was reduce the font size by 2px in comparison to the default. There's other places in the code in which this was done as well.

bennybroseph commented:
It fixes the issue you're having with shadows

bennybroseph commented:
Also, let's not forget to test this in legacy UI

tomasrggm commented:
Looks fine on legacy UI. Checked in all languages. There was never an issue with shadows, it was because I misinterpreted the fontSize property which led me to make a mistake and messed it up myself. 
![image](https://github.com/pagefaultgames/pokerogue/assets/44369723/e2b5744a-24f4-47ab-99ce-5a8783436977)

brain-frog commented:
2 things 
1. can you please update the PR description with the most recent screenshots when visuals change
2. can you show examples of languages that show up differently or have a different font (Chinese, Korean)

tomasrggm commented:
1. Updated the pr as requested
2. Here's examples for chinese and korean. Western languages look the same as english visual wise:
![image](https://github.com/pagefaultgames/pokerogue/assets/44369723/b78d981e-52f8-4a99-b1aa-91d67fd0a442)
![image](https://github.com/pagefaultgames/pokerogue/assets/44369723/caf91dc0-5bd0-4462-a9a6-31fcc9028ec9)
![image](https://github.com/pagefaultgames/pokerogue/assets/44369723/e42f7a97-4fff-4df0-ad85-9fdc3843a738)
The chinese fonts seem needlessly massive in my opinion. Though I think that is out of scope for this PR and should be checked at someplace else

bennybroseph commented:
> Looks fine on legacy UI. Checked in all languages. There was never an issue with shadows, it was because I misinterpreted the fontSize property which led me to make a mistake and messed it up myself.

Assuming this is the latest version, there is definitely some issues with shadows. In general the font is very blurry, but the shadow is barely offset at all and is making it more apparent. Just compare the icon sharpness (which needs some help anyway) to the font and you'll see what I mean

![image](https://github.com/pagefaultgames/pokerogue/assets/13838608/26459e66-e724-4b30-a472-5bc3441fea7b)
![image](https://github.com/pagefaultgames/pokerogue/assets/13838608/a22cebd5-9d36-46f2-8098-543ea036fb97)

tomasrggm commented:
I see your point now. Yes fiddling with the font in a way its original scale isn't preserved is probably what is making it blurry. I've decided to settle with the original size which now that I've downscaled the moves container isn't a bad solution in itself. Here's an example in english and chinese (remaining western/asian languages are about the same):
![image](https://github.com/pagefaultgames/pokerogue/assets/44369723/4202eb04-0c20-43c7-b531-aada100fdc23)
![image](https://github.com/pagefaultgames/pokerogue/assets/44369723/c716a6f6-f73a-4ab9-b6c8-c9a01fc551b4)

tomasrggm commented:
My mistake, I misinterpreted the suggested changes hence why the eslint failed. Should be good to go as intended now.

tomasrggm commented:
Any updates on whether this can be merged or not?

flx-sta commented:
### Are you still working on this PR?
-----
No actions for over a month

tomasrggm commented:
I've addressed all feedback hence why it hasn't been updated since.

flx-sta commented:
> I've addressed all feedback hence why it hasn't been updated since.

All good. I’m just going through PRs to sort out outdated/stale ones. Thanks for the response

flx-sta commented:
@tomasrggm there is some merge-conflicts to resolve

tomasrggm commented:
@flx-sta should be good to go

flx-sta reviewed (approved):
lgtm

## Desktop (DE)
<img width="437" alt="image" src="https://github.com/user-attachments/assets/5139ca2c-3e92-4a06-813c-65a693a596c1">

## Mobile (DE)
<img width="863" alt="image" src="https://github.com/user-attachments/assets/fb3caa20-f3ed-4d9c-8df9-ed24c5614245">
<img width="379" alt="image" src="https://github.com/user-attachments/assets/39f8da2d-cd5f-4797-8199-8f81d53b954b">

DayKev commented on src/ui/stats-container.ts:
```diff
@@ -55,12 +57,13 @@ export class StatsContainer extends Phaser.GameObjects.Container {
 
     for (const s of PERMANENT_STATS) {
-      const statLabel = addTextObject(this.scene, ivChartBg.x + (ivChartSize) * ivChartStatCoordMultipliers[s][0] * 1.325, ivChartBg.y + (ivChartSize) * ivChartStatCoordMultipliers[s][1] * 1.325 - 4 + ivLabelOffset[s], i18next.t(getStatKey(s)), TextStyle.TOOLTIP_CONTENT);
+      const statLabel = addTextObject(this.scene, ivChartBg.x + (ivChartSize) * ivChartStatCoordMultipliers[s][0] * 1.325 + (this.showDiff ? 0 : ivLabelOffset[s]), ivChartBg.y + (ivChartSize) * ivChartStatCoordMultipliers[s][1] * 1.325 - 4 + (this.showDiff ? 0 : ivChartLabelyOffset[s]), i18next.t(getStatKey(s)), TextStyle.TOOLTIP_CONTENT);
```
```suggestion
      const statLabel = addTextObject(
        this.scene,
        ivChartBg.x + (ivChartSize) * ivChartStatCoordMultipliers[s][0] * 1.325 + (this.showDiff ? 0 : ivLabelOffset[s]),
        ivChartBg.y + (ivChartSize) * ivChartStatCoordMultipliers[s][1] * 1.325 - 4 + (this.showDiff ? 0 : ivChartLabelyOffset[s]),
        i18next.t(getStatKey(s)),
        TextStyle.TOOLTIP_CONTENT
      );
```
This line is getting needlessly long.

tomasrggm commented:
@DayKev applied suggestion.

flx-sta commented:
@f-fsantos as previously wanted I did a check on beta and I admit it’s fairly small now but one can still read it (iPhone 13 screen).
That’s my feedback
